src/services/db/GitHubService.js

```diff
@@ -94,6 +94,23 @@ class GitHubService {
     const { accessToken, siteName, isomerUserId: userId } = sessionData
     try {
       const endpoint = this.getFilePath({ siteName, fileName, directoryName })
+      const isCreatingTopLevelDirectory = fileName === "collection.yml"
+      const isCreatingNewResourceFolder = fileName === "index.html"
```

Comment:
just curious, how do we know that if filename===index.html means a new resource folder is created?

Reply:
It's our invariant for repo structure currently, but that might change in future - I think I'm okay with this for now, but we should leave a comment here about the invariants we currently use, so that we can modify this in future if necessary

Reply:
okay, let's leave comments for future context

Reply:
comment added!